### PR TITLE
Zephyr: lwm2m sample

### PR DESCRIPTION
**Pull request by jonas-rem**
Earlier, the zephyr start script started the sample in the Zephyr directory.  Now in order to test better, we have to add different functionality. For that this PR switches to the local Zephyr sample `fw_test/lwm2m_client`.

Update to the most recent version of Zephyr (copy the unmodified lwm2m sample) before adapting it.

Switch to LwM2M 1.1, add resource 5/0/1 to test OTA updates via download link.

### CODE REVIEW

**Review by Kappuccino111 — COMMENTED** [review]
@jonas-rem, this is definitely what I was looking for. The LwM2M client in Zephyr samples is configured only for temperature updates. I was wondering if we could set up our own client for testing other parameters like humidity, pressure, sound , etc. The fw_test directory seemed like a copy of the sample LwM2M client in Zephyr, and I wasn't sure what it did. Now it makes sense.

File: fw_test/lwm2m_client/src/firmware_update.c [hunk 1]
```diff
@@ -58,6 +68,7 @@ void init_firmware_update(void)
 {
 	/* setup data buffer for block-wise transfer */
 	lwm2m_register_pre_write_callback(&LWM2M_OBJ(5, 0, 0), firmware_get_buf);
+	lwm2m_register_post_write_callback(&LWM2M_OBJ(5, 0, 1), firmware_download_cb);
```
Comment (Kappuccino111) [inline fw_test/lwm2m_client/src/firmware_update.c:71]: @jonas-rem is this for the bi-directional communication ?? The automatic firmware update that would happen if the user uploads a binary from Django.

Also what exactly is (5,0,1) ??

**Comment by Kappuccino111** [comment]
@jonas-rem also are all files updated in this PR currently being changed to handle the bidirectional update part ? Because mostly this is just the initial lwm2m client from the Zephyr Sample.

File: fw_test/lwm2m_client/src/firmware_update.c (hunk 1 above)
Comment (jonas-rem) [reply fw_test/lwm2m_client/src/firmware_update.c:71]: Yes, exactly. 5/1/1 is a link (URL String) to a firmware binary. Zephyr will download the file, report that and wait for further instructions. Ultimately Zephyr will write this file to a b Partition, perform a reset and boot to the new firmware.

**Comment by jonas-rem** [comment]
> @jonas-rem also are all files updated in this PR currently being changed to handle the bidirectional update part ? Because mostly this is just the initial lwm2m client from the Zephyr Sample.
> 
> If that is the case then can we change this to a draft PR ? Or is this complete ?

With this current version we can receive the Update URL Resource from the LwM2M Server. In addition it's a major overhaul that syncs with the most recent version of Zephyr. In addition it enables 

Handling queue mode (being offline most of the time) is something that is missing in the server as well as in the firmware. Not sure if a true firmware update will work in the simulation, but we can give it a try.

We can keep it as draft for now.

**Comment by jonas-rem** [comment]
> @jonas-rem, this is definitely what I was looking for. The LwM2M client in Zephyr samples is configured only for temperature updates. I was wondering if we could set up our own client for testing other parameters like humidity, pressure, sound , etc. The fw_test directory seemed like a copy of the sample LwM2M client in Zephyr, and I wasn't sure what it did. Now it makes sense.

Sure we can test several things with it. We could also create multiple clients to test different scenarios in an automated way.

**Review by Kappuccino111 — APPROVED** [review]
Setup for a custom LwM2M client. This adds client config for work on bidirectional communication. Good to go 👍

**Pull request merged**